### ISSUE TITLE
`GetLanguageInfo()->CanonicalName` not as expected

### ISSUE THREAD

**Issue by 0tkl** [TypesettingTools/Aegisub#211]
https://github.com/TypesettingTools/Aegisub/blob/65e7fbe5b35056da1e3596605ae1494bc74e40d5/src/aegisublocale.cpp#L96-L102

Install Aegisub on a ko-KR locale Windows computer, and open Aegisub for the first time. The Korean language does not appear at the top of the list.

![캡처1](https://github.com/user-attachments/assets/6f4d84c1-b3fa-4a43-8530-caf3ddb30a99)

**Comment by KerimDemirkaynak** [comment]
Turkish language is also not at the top of the list.

![image](https://github.com/user-attachments/assets/e02b0f73-b0ba-4e73-b2fe-82fd9412b4dd)

**Comment by 0tkl** [comment]
Having tested it for a bit, I now suspect that `AegisubLocale::HasLanguage()` is working correctly, so `AegisubApp::OnInit()` is likely able to correctly select the appropriate UI language. The bug probably lies solely within `AegisubLocale::PickLanguage()`.

https://github.com/TypesettingTools/Aegisub/blob/a5a7bc58f2552a2ee2283c445d6e7c68e57d898c/src/main.cpp#L230-L236

**Comment by arch1t3cht** [comment]
What does `wxLocale::GetLanguageInfo(wxLocale::GetSystemLanguage())->CanonicalName` return for you? Does it return `ko` or `ko_KR`?

**Comment by arch1t3cht** [comment]
FWIW I cannot reproduce this after setting my system language to Turkish (`wxLocale::GetLanguageInfo(wxLocale::GetSystemLanguage())->CanonicalName` being `tr_TR`) since Aegisub will not even show the language choice dialog since `GetBestTranslation` returns an existing translation. It's quite possible that wxWidgets's behavior for `GetBestTranslation` changed since this issue was opened (I know for a fact that it changed on master, just not if the changes were also backported to 3.2) - can you check if you still get a language selection prompt on newer builds?

(Of course, this does not change the fact that the language selection prompt, if it does show up, does not suggest the languages it maybe should, but if the language autoselection works correctly in all relevant cases then that would no longer be too much of an issue.)

**Comment by KerimDemirkaynak** [comment]
> FWIW I cannot reproduce this after setting my system language to Turkish (`wxLocale::GetLanguageInfo(wxLocale::GetSystemLanguage())->CanonicalName` being `tr_TR`) since Aegisub will not even show the language choice dialog since `GetBestTranslation` returns an existing translation. It's quite possible that wxWidgets's behavior for `GetBestTranslation` changed since this issue was opened (I know for a fact that it changed on master, just not if the changes were also backported to 3.2) - can you check if you still get a language selection prompt on newer builds?
> 
> (Of course, this does not change the fact that the language selection prompt, if it does show up, does not suggest the languages it maybe should, but if the language autoselection works correctly in all relevant cases then that would no longer be too much of an issue.)

No, the language selection dialog no longer appears on first launch. However, if I open it manually, the system language is still not moved to the top of the list.

**Comment by 0tkl** [comment]
Aegisub 3.5.0 beta, tested on a freshly installed Windows 11 virtual machine with `ko-KR` locale. The problem still exists.

<img width="1842" height="898" alt="Image" src="https://github.com/user-attachments/assets/e3e52765-534a-4359-93f3-033e87ccf640" />
